app/forms/candidate_interface/fac_invite_response_form.rb

```diff
@@ -0,0 +1,36 @@
+module CandidateInterface
+  class FacInviteResponseForm
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attr_reader :application_choice
+
+    attribute :application_form
+    attribute :invite
+    attribute :apply_for_this_course, :string
+
+    validates :apply_for_this_course, presence: true
+
+    def save
+      return false if invalid?
+
+      course_option = CourseOption.find_by(course_id: invite.course_id)
```

Comment:
A course can have many options, so this is just picking the first one. That might be fine (the candidate can edit it before submitting), but it won't be fine if the provider has decided the the site isn't selectable. If it's not selectable, we have to make sure we find the course_option for their main site.

Reply:
We need a bit of input from product on this. I like the idea of selecting the first site.

I don't think we have a concept of a main site. We usually select the first one if the provider doesn't allow the candidate to choose.

There's a boolean field on the providers table, `selectable_school` that decides this.

I think we should default to the first course_option/site if provider.selectable_school is true.

Reply:
```suggestion     
      course_option = CourseOption.find_by(course_id: invite.course_id)

      @application_choice = application_form.application_choices.new
      @application_choice.tap do |choice|
        ActiveRecord::Base.transaction do
          choice.configure_initial_course_choice!(course_option)

          choice.update(school_placement_auto_selected: !invite.provider.selectable_school?)
        end
      end
```

This should allow the candidate to select a different site on the draft application page if provider has selectable_school true.

It will also update the invite candidate_decision to applied, by calling `choice.configure_initial_course_choice!(course_option)`

Which eventually calls this https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_choice.rb#L272-L277

It will also set the current_course_option and course_option.